test/tests/functional/pbs_qrun.py

```diff
@@ -128,14 +130,14 @@ def test_qrun_hangs(self):
             except PbsRunError as e:
                 self.logger.info("Runjob throws error: " + e.msg[0])
         else:
-            out = self.scheduler.log_match(
-                "Starting Scheduling Cycle",
-                allmatch=True, interval=5, starttime=now, max_attempts=10)
-            if len(out) < 2:
+            try:
+                self.scheduler.log_match("Starting Scheduling Cycle",
+                                         interval=5, starttime=now,
+                                         max_attempts=10)
+                self.logger.info("No hangs. Parent process exit")
+            except PtlLogMatchError:
                 os.kill(pid, signal.SIGKILL)
                 os.waitpid(pid, 0)
                 self.logger.info("Runjob hung. Child process exit.")
                 self.assertFalse(True,
```

Comment:
please use 'self.fail' instead of doing this

Reply:
self.fail() this will fail the PTL test case. But what if the child process hung, then this test case will wait for timeout period runjob. May be instead of waiting that longer, after checking few times in the the sched log if new cycle not started we can kill the child, as we know it didnt start the cycle and test case hung.

Reply:
i agree with killing the child process, I just wanted you to replace the self.assertFalse(True, .. ) with self.fail(...)